### ISSUE TITLE
[docker] Add docker images for linux/aarch64 platform

### ISSUE THREAD

**Issue by krfricke** [ray-project/ray#31537]
### Description

https://github.com/ray-project/ray/pull/31522 added ARM64 wheel builds to CI. However, as discussed in the PR and in #28103, ideally users can use `rayproject/ray` images directly.

We should thus add a docker image build step to CI that builds the linux/aarch64 docker images.

The main open question (at least for me) is how we make sure the platform-specific images are kept in sync and added to the potentially existing images from other platforms. Concretely:

- `rayproject/ray:commithash` for `linux/x86_64` and `linux/aarch64` should be the same
- Do we allow `rayproject/ray:nightly` for `linux/x86_64` and `linux/aarch64` to point to different commits?
- Does uploading an aarch64 image to dockerhub "add to" or _replace_ an existing x86_64 image?

**Comment by KastanDay** [comment]
For others searching about this issue, my error was 
```
Error response from daemon: image with reference my-ray-image:latest was found but does not match the specified platform: wanted linux/arm64, actual: linux/amd64
```
Thank you for your efforts.

**Comment by krfricke** [comment]
We've just merged the build job for arm64 docker images with a tag suffix: https://github.com/ray-project/ray/issues/31537

This means you can now use e.g. `rayproject/ray:nightly-aarch64` on arm machines.

We haven't released images for Ray 2.2.0 - first release with Docker images will be Ray 2.3.0 (targeted for mid february)

See here for all available arm64 images: https://hub.docker.com/r/rayproject/ray/tags?page=1&name=aarch64

And here for a roadmap RFC for our arm64 docker images: https://github.com/ray-project/ray/issues/31966
